pkg/agent/packetcapture/packetcapture_controller.go

```diff
@@ -607,117 +648,223 @@ func (c *Controller) uploadPackets(ctx context.Context, pc *crdv1alpha1.PacketCa
 	if err != nil {
 		return fmt.Errorf("failed to generate SSH client config: %w", err)
 	}
-	return uploader.Upload(pc.Spec.FileServer.URL, c.generatePacketsPathForServer(pc.Name), cfg, outputFile)
+	localFileName := fmt.Sprintf("%s-%s-%s", pc.Name, location, env.GetPodName())
+	return uploader.Upload(pc.Spec.FileServer.URL, c.generatePacketsPathForServer(localFileName), cfg, outputFile)
+}
+
+func getFirstPath(paths []string) string {
+	if len(paths) > 0 {
+		return paths[0]
+	}
+	return ""
 }
 
 func (c *Controller) updateStatus(ctx context.Context, pc *crdv1alpha1.PacketCapture, state packetCaptureState) error {
-	// Make a deepcopy as the object returned from lister must not be updated directly.
-	toUpdate := pc.DeepCopy()
-	var conditions []crdv1alpha1.PacketCaptureCondition
-	t := metav1.Now()
-	desiredStatus := crdv1alpha1.PacketCaptureStatus{
-		NumberCaptured: state.capturedPacketsNum,
-		FilePath:       state.filePath,
-	}
-
-	var conditionStarted, conditionComplete, conditionUploaded crdv1alpha1.PacketCaptureCondition
-	switch state.phase {
-	case packetCapturePhasePending:
-		if state.captureErr != nil {
-			conditionStarted = crdv1alpha1.PacketCaptureCondition{
-				Type:               crdv1alpha1.PacketCaptureStarted,
-				Status:             metav1.ConditionStatus(v1.ConditionFalse),
-				LastTransitionTime: t,
-				Reason:             "NotStarted",
-				Message:            state.captureErr.Error(),
+	if retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		// Get the latest version of the PacketCapture to avoid working with a stale object.
+		// This is crucial for handling concurrent updates from different agents.
+		latestPC, err := c.crdClient.CrdV1alpha1().PacketCaptures().Get(ctx, pc.Name, metav1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return nil
 			}
-		} else {
+			return err
+		}
+		toUpdate := latestPC.DeepCopy()
```

Comment:
if you are getting the new object everytime, seems no need to do the DeepCopy anymore?